release-static failing

### Which jobs are failing?

release-static

### Which tests are failing?

N/A

### Since when has it been failing?

Not sure, just noticed this morning

### Reason for failure (if possible)

```
  = note: "cc" "-m64" "/home/runner/work/conmon-rs/conmon-rs/target/x86_64-unknown-linux-gnu/release/deps/conmonrs-988be6b2ab5ecfbe.conmonrs-4c05c21a52340583.3ioxwzlr70pbvyba.rcgu.o.rcgu.o" "-Wl,--as-needed" "-L" "/home/runner/work/conmon-rs/conmon-rs/target/x86_64-unknown-linux-gnu/release/deps" "-L" "/home/runner/work/conmon-rs/conmon-rs/target/release/deps" "-L" "src/backend/linux_raw/arch/outline/release" "-L" "/home/runner/work/conmon-rs/conmon-rs/target/x86_64-unknown-linux-gnu/release/build/libgit2-sys-adb848e1e8fe704a/out/build" "-L" "/home/runner/.rustup/toolchains/1.58.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/tmp/rustcOeSN7b/liblibgit2_sys-9c0b940abda9ef93.rlib" "/tmp/rustcOeSN7b/librustix-371ab8aa59bbc384.rlib" "-Wl,--start-group" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-lgcc_eh" "-lgcc" "-Wl,--end-group" "/home/runner/.rustup/toolchains/1.58.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-5667a4a7e2c48d47.rlib" "-Wl,-Bdynamic" "-lc" "-lz" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/home/runner/.rustup/toolchains/1.58.1-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/home/runner/work/conmon-rs/conmon-rs/target/x86_64-unknown-linux-gnu/release/deps/conmonrs-988be6b2ab5ecfbe" "-Wl,--gc-sections" "-static" "-no-pie" "-Wl,-zrelro,-znow" "-nodefaultlibs"
  = note: /usr/bin/ld: /home/runner/work/conmon-rs/conmon-rs/target/x86_64-unknown-linux-gnu/release/deps/conmonrs-988be6b2ab5ecfbe.conmonrs-4c05c21a52340583.3ioxwzlr70pbvyba.rcgu.o.rcgu.o: in function `<std::sys_common::net::LookupHost as core::convert::TryFrom<(&str,u16)>>::try_from':
          /rustc/db9d1b20bba1968c1ec1fc49616d4742c1725b4b//library/std/src/sys_common/net.rs:191: warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
          /usr/bin/ld: attempted static link of dynamic object `/lib/x86_64-linux-gnu/libc.so.6'
          /usr/bin/ld: attempted static link of dynamic object `/lib64/ld-linux-x86-64.so.2'
          /usr/bin/ld: attempted static link of dynamic object `/usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libz.so'
          collect2: error: ld returned 1 exit status
          

```

### Anything else we need to know?

ref https://github.com/containers/conmon-rs/actions/runs/3675616095/jobs/6218022389

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -72,7 +72,10 @@ jobs:
       - name: Select Toolchain
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: ${{ env['ACTION_MSRV_TOOLCHAIN']  }}
+          # Ubuntu 22.04 glibc static is not compatible with rustc 1.58.1 (see
+          # ACTION_MSRV_TOOLCHAIN). Means we now just use the latest one, since
+          # the static builds are made for the community.
+          toolchain: stable
           default: true
           override: true
           components: rustfmt

